api/v1alpha1/shared_types.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2023.
```

Comment:
nit: should it be 2024 now? :)

Reply:
I believe this came up in Numaflow as well and the decision was to keep it the same year. I can see some files in numaflow that still say 2022 even though they're newer than that.